Sources/SwiftDocC/Model/Rendering/RenderContentCompiler.swift

```diff
@@ -165,7 +165,18 @@ struct RenderContentCompiler: MarkupVisitor {
             return [RenderInlineContent.text(link.plainText)]
         }
         
-        return [RenderInlineContent.reference(identifier: .init(resolved.absoluteString), isActive: true, overridingTitle: nil, overridingTitleInlineContent: nil)]
+        let linkTitleInlineContent = link.children.flatMap { visit($0) } as! [RenderInlineContent]
+        let plainTextLinkTitle = linkTitleInlineContent.plainText
+        let overridingTitle = plainTextLinkTitle.isEmpty ? nil : plainTextLinkTitle
+        let overridingTitleInlineContent = linkTitleInlineContent.isEmpty ? nil : linkTitleInlineContent
+        return [
+            RenderInlineContent.reference(
+                identifier: .init(resolved.absoluteString),
+                isActive: true,
+                overridingTitle: link.isAutolink ? nil : overridingTitle,
+                overridingTitleInlineContent: link.isAutolink ? nil : overridingTitleInlineContent
+            )
+        ]
```

Comment:
A possible solution to fix the failed test case and remove the need of Swift-Markdown's support for `Markdown.Link._isAutolink` property / `Markdown.Link.isAutolink` setter
```suggestion
        let useOverriding: Bool
        if link.isAutolink { // If the link is an auto link, we don't use overriding info
            useOverriding = false
        } else if let overridingTitle,
                  overridingTitle.hasPrefix(DocumentationSchemeHandler.scheme + ":"),
                  destination.hasPrefix(DocumentationSchemeHandler.fullScheme),
                  destination.hasSuffix(overridingTitle.dropFirst((DocumentationSchemeHandler.scheme + ":").count)) { // If the link is a transformed doc link, we don't use overriding info
            useOverriding = false
        } else {
            useOverriding = true
        }
        return [
            RenderInlineContent.reference(
                identifier: .init(resolved.absoluteString),
                isActive: true,
                overridingTitle: useOverriding ? overridingTitle : nil,
                overridingTitleInlineContent: useOverriding ? overridingTitleInlineContent : nil
            )
        ]
```